hibernate-core/src/main/java/org/hibernate/jpa/internal/PersistenceUnitUtilImpl.java

```diff
@@ -68,7 +68,7 @@ public Object getIdentifier(Object entity) {
 		}
 
 		if ( entity instanceof HibernateProxy ) {
-			return ((HibernateProxy) entity).getHibernateLazyInitializer().getIdentifier();
+			return ((HibernateProxy) entity).getHibernateLazyInitializer().getInternalIdentifier();
```

Comment:
l'm not sure about the above change, since the method implements `PersistenceUnitUtil#getIdentifier`. I'll check the spec tomorrow.

Reply:
I'll also run the TCK tonight.

Reply:
Looks like we got stuck here - @gbadner could you suggest how to proceed please?

Reply:
Sorry about that. I ran the TCK with this change and there were no failures.